[autodiscovery/terragrunt] empty terraform block breaks updatecli

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

If you have a terragrunt file specifying a terraform block without setting source in it, it segfault updatecli

### Expected Behavior

Updatecli should just ignore the manifest

### Steps To Reproduce

A project with the following kind of terragrunt manifest
```hcl
terraform {}

locals {}

generate "main" {
  path      = "main.tf"
  if_exists = "overwrite_terragrunt"
  contents  = <<EOF
variable "tags" {
  type = map(string)
  default = {}
}
EOF
}
```

And the following compose policy
```yaml
policies:
  - name: Terragrunt autodiscovery
    policy: ghcr.io/updatecli/policies/autodiscovery/terragrunt:0.3.0@sha256:976dc3ac101e19962e4c6941dfd145ccdbd40b8f6c87ac697cd1d422607e6072

```

Would yield
```
TERRAGRUNT
===========
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x6d414ae]

```

### Environment

```markdown
- OS:
- updatecli:
```


### Anything else?

_No response_

--- a/pkg/plugins/autodiscovery/terragrunt/test/testdata/terraform_without_source.hcl
+++ b/pkg/plugins/autodiscovery/terragrunt/test/testdata/terraform_without_source.hcl
@@ -1,0 +1,14 @@
+terraform {}
+
+locals {}
+
+generate "main" {
+  path      = "main.tf"
+  if_exists = "overwrite_terragrunt"
+  contents  = <<EOF
+variable "tags" {
+  type = map(string)
+  default = {}
+}
+EOF
+}

--- a/pkg/plugins/autodiscovery/terragrunt/utils.go
+++ b/pkg/plugins/autodiscovery/terragrunt/utils.go
@@ -64,7 +64,11 @@ func getTerragruntModule(filename string, allowNoVersion bool) (module *terragru
 
 	for _, block := range hclfile.Body().Blocks() {
 		if block.Type() == "terraform" {
-			quotedSource := strings.TrimSpace(string(block.Body().GetAttribute("source").Expr().BuildTokens(nil).Bytes()))
+			sourceBlock := block.Body().GetAttribute("source")
+			if sourceBlock == nil {
+				return nil, nil
+			}
+			quotedSource := strings.TrimSpace(string(sourceBlock.Expr().BuildTokens(nil).Bytes()))
 			source, hclContext, err := evaluateHcl(quotedSource, data, filename)
 			if err != nil {
 				return nil, err

--- a/pkg/plugins/autodiscovery/terragrunt/utils_test.go
+++ b/pkg/plugins/autodiscovery/terragrunt/utils_test.go
@@ -26,6 +26,7 @@ func TestSearchTerragruntFiles(t *testing.T) {
 				"test/testdata/more_complex_localized.hcl",
 				"test/testdata/non_tfr.hcl",
 				"test/testdata/simple_localized.hcl",
+				"test/testdata/terraform_without_source.hcl",
 				"test/testdata/terragrunt.hcl",
 			},
 		},
@@ -155,6 +156,11 @@ func TestGetTerragruntModules(t *testing.T) {
 					sourceType:      SourceTypeGit,
 				},
 			},
+		},
+		{
+			name:           "Terragrunt terraform block without source",
+			file:           "test/testdata/terraform_without_source.hcl",
+			expectedModule: nil,
 		},
 	}
 

